### ISSUE TITLE
Question: what does subtract 1 month actually mean?

### ISSUE THREAD

**Issue by avand** [iamkun/dayjs#2375]
I've tried looking at the docs and reading the source but I can't actually make sense of what it means to "subtract a month" from a dayjs object. I thought it meant "the same date in the preceding month" but that doesn't seem to hold true in practice and doesn't make sense for dates like the 31st, where the preceding month doesn't have a 31st. Does it equivalent to `30, 'days'`?

Thanks in advance!

**Comment by kito-arch** [comment]
Hi,
I experimented with different values. Here is what I found:

If you have a value of date as 31, considering the month can have 31 days, for example: `2023-10-31`. Subtracting a month from this by `dayjs(2023-10-31, "YYYY-DD-MM")` would result in `2023-09-30`.

But if you have value like `2023-09-30`, Subtracting a month from this by `dayjs(2023-09-30, "YYYY-DD-MM")` would result in `2023-08-30` even though 8th month can have 31 days.

Here is the code I found, responsible for this behavior:
`this.$d = date.set(C.DATE, Math.min(this.$D, date.daysInMonth())).$d`
Let me explain:
One thing is, we are basically subtracting the month by `getMonth() - 1` this would result in subtraction of month but keep the original date even it is 31. So if we subtract month from `2023-10-31` would result in `2023-09-31`. And we know 9th month cannot have 31 as a date.
Second thing is, we are cloning the same date object as above after month subtraction, i.e, `2023-09-31` and we calling a `daysInMonth()` method on this object.
Finally we taking the minimum of both values and assigning it to our object.

Hope you understood.
